### ISSUE TITLE
2D Chromosome grid is drawn over the colormap and matrix label

### ISSUE THREAD

**Issue by Chadius** [higlass/higlass#490]
#### Steps to reproduce
1. Create a view config with a center track. The track's contents contain a heatmap and a chromosome size layer. It doesn't matter if you add the chromosome track before or after adding the heatmap.
2. View the view config in the Higlass Display.

#### Observed behavior
- Before adding the chromosome grid:
![image 4](https://user-images.githubusercontent.com/1376213/50245589-e13c8000-03a0-11e9-84c2-c208c71551a0.png)

- After adding the grid, it obscures the heatmap. 
![image 3](https://user-images.githubusercontent.com/1376213/50245584-df72bc80-03a0-11e9-8cb9-193568ee128f.png)

- If you zoom in, the grid isn't a problem, as it fits inbetween the gaps.
![image 2](https://user-images.githubusercontent.com/1376213/50245574-dd106280-03a0-11e9-80bd-eccb7bf64f53.png)

#### Expected behavior
- The chromosome grid is drawn underneath the heatmap

**Comment by Chadius** [comment]
Could you draw the grid underneath the color scale? If you see the second image, the grid is drawn on top of the scale, which it shouldn't. I'll make a mockup.

**Comment by Chadius** [comment]
![clean-overlay](https://user-images.githubusercontent.com/1376213/51276320-61ef9b80-19a2-11e9-8c09-4b181c86ed46.png)
As an outcome, I'd like to at least see the scale unobstructed. Compare the upper right corner of this mockup to the second picture from the top, where the grid lines are obstructing the scale.

**Comment by Chadius** [comment]
Yes, exactly. Sorry about mixing the images up when I posted initially.

**Comment by flekschas** [comment]
Cool and no worries! And thanks for clarifying :) This is clearly an issue. I'll change the title of the issue slightly and, hopefully, we get around implementing it sometime soon.

**Comment by pkerpedjiev-zymergen** [comment]
This is going to be tricky to implement. The problem is that the heatmap and its color map are on one layer and the grid lines are on another. There's a few options for what we can do:

1. Add an option to display grid lines natively in the heatmap track. That chromosome information is available in the cooler file and we can pass it over in the tileset info.
2. Create an option to hide the actual heatmap and only display the colorscale. That we can add two heatmaps, hide the color scale in the first and the heatmap in the second and sandwich the grid lines in between.

The first is probably slightly more elegant but the second is easier to implement.

**Comment by flekschas** [comment]
Do you think we could somehow broadcast the location of a drawn label and the colormap? Then we could add a mask to the chrom grid. The new overlay chrom grid already needs to apply masks so it wouldn't be hard to extend this.

**Comment by pkerpedjiev-zymergen** [comment]
I'd prefer not to do that. It's introducing a whole new level of complexity that will be hard to maintain and keep track of.

**Comment by flekschas** [comment]
That's true. I'd add quite a bit of complexity
